src/source/source_simple_led.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  source_component_base,
+  type SourceComponentBase,
+} from "src/source/base/source_component_base"
+import { z } from "zod"
+import { expectTypesMatch } from "src/utils/expect-types-match"
+
+export const source_simple_led = source_component_base.extend({
+  ftype: z.literal("simple_led"),
```

Comment:
things to add:

`color` (optional string)
`wavelength` (optional string)

all the parameters of diode (in fact this should extend diode)